contributor_docs/p5.strands.md

```diff
@@ -0,0 +1,251 @@
+4. p5.strands generates GLSL code from that graph.
+
+## Why pseudo-JavaScript?
+
+The code the user writes when using p5.strands is mostly JavaScript, with some extensions. Shader code heavily encourages use of vectors, and the extensions all make this as easy in JavaScript as in GLSL.
+- In JavaScript, there is not a vector data type. In p5.strands, you create vectors by creating array, e.g. `myVec = [1, 0, 0]`. You can't use actual arrays in p5.strands; all arrays are fixed-size vectors.
+- In JavaScript, you can only use mathematical operators like `+` between numbers and strings, not with vectors. In p5.strands, we allow use of these operators between vectors.
+- In GLSL, you can do something called *swizzling*, where you can create new vectors out of the components of an existing vector, e.g. `myvec.xy`, `myvec.bgr`, or even `myvec.zzzz`. p5.strands adds support for this on its vectors.
+
+When we transpile the input code, we rewrite these into valid JavaScript. Array literals are turned into function calls like `vec3(1, 0, 0)` which return vector class instances. These instances are wrapped in a `Proxy` that handles property accesses that look like swizzles, and converts them into sub-vector references. Operators between vectors like `a + b` are rewritten into method calls, like `a.add(b)`. 
+
+If a user writes something like this:
+
+```js
+baseMaterialShader().modify(() => {
+  const t = uniformFloat(() => millis())
+  getWorldInputs((inputs) => {
+    inputs.position += [20, 25, 20] * sin(inputs.position.y * 0.05 + t * 0.004)
+    return inputs
+  })
+})
+```
+
+...it gets transpiled to something like this:
+```js
+baseMaterialShader().modify(() => {
+  const t = uniformFloat('t', () => millis())
+  getWorldInputs((inputs) => {
+    inputs.position = inputs.position.add(strandsNode([20, 25, 20]).mult(sin(inputs.position.y.mult(0.05).add(strandsNode(t).mult(0.004)))))
+    return inputs
+  })
+})
+```
+
+## The program graph
+
+The overall structure of a shader program is represented by a **control-flow graph (CFG)**. This divides up a program into chunks that need to be outputted in linear order based on control flow. A program like the one below would get chunked up around the if statement:
+
+```js
+// Start chunk 1
+let a = 0;
+let b = 1;
+// End chunk 1
+
+// Start chunk 2
+if (a < 2) {
+  b = 10;
+}
+// End chunk 2
+
+// Start chunk 3
+b += 2;
+return b;
+// End chunk 3
+```
+
+We store the individual states that variables can be in as nodes in a **directed acyclic graph (DAG)**. This is a fancy name that basically means each of these variable states may depend on previous variable states, and outputs can't feed back into inputs. Each time you modify a variable, that represents a new state of that variable. For example, below, it is not sufficient to know that `c` depends on `a` and `b`; you also need to know *which version of `b`* it branched off from:
```

Comment:
This whole documentation is wonderful! Very minor and not necessary for this iteration: DAG illustration / connecting it to code block below not only in text